binary_search_tree/tree.py

```diff
@@ -7,51 +7,152 @@ def __init__(self, key, val = None):
         self.value = val
         self.left = None
         self.right = None
-        
-
+    
 
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
+    # Time Complexity: O(n)
+    # Space Complexity: O(1)
     def add(self, key, value = None):
```

Comment:
👍  However the time complexity is O(n) for an unbalanced tree and O(log n) for a balanced tree.

Also the space complexity is the same due to the call stack (recursion).